packages/compass-preferences-model/src/preferences.ts

```diff
@@ -168,6 +169,19 @@ function deriveReadOnlyOptionState<K extends keyof AllPreferences>(
   });
 }
 
+/** Helper for defining how to derive value/state for protectConnectionStrings-affected preferences */
+function deriveProtectConnectionStringsOptionState<
+  K extends keyof AllPreferences
+>(property: K): DeriveValueFunction<boolean> {
+  return (v, s) => ({
+    value: v(property) && !v('protectConnectionStrings'),
+    state:
+      s(property) ??
+      s('protectConnectionStrings') ??
+      (v('protectConnectionStrings') ? 'derived' : undefined),
+  });
```

Comment:
I’m wondering if this should account for more than just `protectConnectionStrings`, since devtools can effectively be used to do anything with Compass (i.e. basically everything in the `Features` tab)? On the other hand, we probably still want to be able to inspect code while debugging.

The tech design lists this as an option that cannot be set in the UI. If we go that way, we can use the default of `false` here as it is, and require users (like us) to explicitly opt into enabling dev tools through the command line and/or global config file. Is that an approach we want to take? (This might be a separate concern, and we probably still want to keep this logic here, since `protectConnectionStrings: true` in the global config file should probably not be something a user can override through `--enableDevTools` on the command line.)

Reply:
Yes, I missed this point for enableShell as well, the setting should not be editable from UI according to scope. But is it so bad tho? I mean if these features also like readOnly should not be editable for users they can be disabled via global setting or cli args. And if they are not disabled at those levels then we can still keep enableDevTools as default false as it is right now.

Reply:
Yeah, I don’t think it’s bad for enableShell really and it’s probably not bad here either. It just means we need to be careful about disabling this option when _any_ other option that is used to lock Compass down is set.